qcore/geo.py

```diff
@@ -18,33 +19,59 @@ class InputError(Exception):
 
 
 def get_distances(locations: np.ndarray, lon: float, lat: float):
-    """Calculates the distance between the array of locations and
+    """
+    Calculates the distance between the array of locations and
     the specified reference location
+    Does the same as get_multiple_distances except for a single reference location
+    Note: Is not within the same function due to numba and type unification
 
     Parameters
     ----------
     locations : np.ndarray
         List of locations
         Shape [n_locations, 2], column format (lon, lat)
     lon : float
-        Longitude of the reference location
-    lat
-        Latitude of the reference location
+        Singular float of a Longitude reference location to compare
+    lat : float
+        Singular float of a Latitude reference locations to compare
 
     Returns
     -------
     np.ndarray
         The distances, shape [n_locations]
     """
+    return get_multiple_distances(locations, lon, lat)[0]
+
+
+@njit
+def get_multiple_distances(locations: np.ndarray, lon: np.float, lat: np.float):
```

Comment:
Whats the difference between the two functions? Just combine them into one?
Also does @njit actually do much in this case since its vectorized?

Reply:
One returns a vector of vectors. One returns a vector. Just so get_distances returns the same datastructure for the same inputs (for compatibility reasons)

I'll let Joel answer the performance question - but I'd expect that since the original had a decent improvement with njit that it still would have that improvement.

Reply:
So turns out that the njit doesn't do much in terms of performance tested with multiple faults averaging 20 times.
![image](https://user-images.githubusercontent.com/33056240/159189212-22dd793b-ebf5-4930-af27-f19b66a04af6.png)

This could be removed, except then would have to go through GMHazard, IM_calc to remove the numba.njit calls to the get_distances function. It is easier to leave it, but willing to do either.

Thoughts @claudio525 ?

Reply:
Now that their is no numba dependence, thoughts on combining these two functions? @jasonmotha @joelridden @jpa198

Reply:
I mean you can do one function called get_distances that has a split return type depending if sites is a vector or not.

I think what we have at the moment is the most elegant soln. though.

Reply:
I think they should combine and have split return types. get_distances feels like its taking alot of undeeded lines up

Reply:
How come this was resolved? 
I def think that they should be combined, two functions for the same thing seems overkill. And its not like this is uncommon, e.g. numpy functions return a float if you pass floats in, but if you pass arrays in then you get arrays back